File: rs/Euclidean_Algorithm.rs
use std::io;
use std::cmp;

fn main() {
    let mut s = String::new();
    match io::stdin().read_line(&mut s) {
        Err(_) => panic!("failed to read line"),
        _ => {},
    }

    let mut ab = [0; 2];
    for (i, x) in s.trim().split(' ').enumerate() {
        ab[i] = match x.parse() {
            Ok(n) => n,
            Err(_) => panic!("not a number"),
        };
    }

    println!("{}", euclidean(cmp::max(ab[0], ab[1]), cmp::min(ab[0], ab[1])));
}

fn euclidean(a: u64, b: u64) -> u64 {
    let m = a % b;
    if m == 0 {
        b
    }
    else {
        euclidean(b, m)
    }
}

File: swift/Euclidean_Algorithm.swift
import Swift

let ab = readLine()!.characters.split(" ").map {
    Int(String($0))!
}

func euclidean(a: Int, b: Int) -> Int {
    switch a % b {
    case 0:
        return b
    case let m:
        return euclidean(b, b: m)
    }
}

print(euclidean(max(ab[0], ab[1]), b: min(ab[0], ab[1])))
